[css-counter-styles] Port ethiopic counter styles to new codepath

nt1m opened the pull request:
#### 08374bddfb52a8231e7b4876949b3c2b63822386
<pre>
[css-counter-styles] Port ethiopic counter styles to new codepath
<a href="https://bugs.webkit.org/show_bug.cgi?id=256158">https://bugs.webkit.org/show_bug.cgi?id=256158</a>
rdar://108724140

Reviewed by Myles C. Maxfield.

Use converter at <a href="https://bugs.webkit.org/attachment.cgi?id=466153">https://bugs.webkit.org/attachment.cgi?id=466153</a> to generate counter styles based on RenderListMarker data.
Note that this is supposed to be a 1:1 port from RenderListMarker, so there are places where we differ from the W3C ready-made counter styles.
rdar://108704382 has been filed to review them.

Also isCounterStyleUnsupportedByUserAgent is now removed and merged back to its call-site.

* Source/WebCore/css/CSSCounterStyleRegistry.cpp:
(WebCore::isCounterStyleUnsupportedByUserAgent): Deleted.
* Source/WebCore/css/CSSCounterStyleRegistry.h:
* Source/WebCore/css/counterStyles.css:
(@counter-style afar):
(@counter-style ethiopic-halehame-ti-er):
(@counter-style ethiopic-halehame-ti-et):
(@counter-style oromo):
(@counter-style sidama):
(@counter-style tigre):
(@counter-style urdu):
(@counter-style amharic):
(@counter-style amharic-abegede):
(@counter-style ethiopic):
(@counter-style ethiopic-abegede):
(@counter-style somali):
(@counter-style tigrinya-er-abegede):
(@counter-style tigrinya-et-abegede):
(@counter-style ethiopic-abegede-gez):
(@counter-style ethiopic-abegede-am-et):
(@counter-style ethiopic-abegede-ti-er):
(@counter-style ethiopic-abegede-ti-et):
(@counter-style ethiopic-halehame-aa-et):
(@counter-style ethiopic-halehame-aa-er):
(@counter-style ethiopic-halehame-am-et):
(@counter-style ethiopic-halehame-gez):
(@counter-style ethiopic-halehame-om-et):
(@counter-style ethiopic-halehame-sid-et):
(@counter-style ethiopic-halehame-so-et):
(@counter-style ethiopic-halehame-tig):
(@counter-style tigrinya-er):
(@counter-style tigrinya-et):
* Source/WebCore/style/StyleBuilderConverter.h:
(WebCore::Style::BuilderConverter::convertListStyleType):

Canonical link: <a href="https://commits.webkit.org/263602@main">https://commits.webkit.org/263602@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a45c5f81efc1224097b8f2701e10f013a2e9b78c

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/5191 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/5325 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/5507 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/6718 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/5261 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/5188 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/5535 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/5303 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/5570 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/5284 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/15/builds/5373 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/4651 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/6735 "Built successfully") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/39/builds/2858 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/4649 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/10324 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/4723 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/4722 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/6341 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/7/builds/5152 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/4240 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/30/builds/4630 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [  ~~🛠 🧪 merge~~](https://ews-build.webkit.org/#/builders/19/builds/1245 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/4/builds/8713 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/584 "Built successfully and passed tests") | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/31/builds/4995 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

vitorroriz commented on Source/WebCore/css/counterStyles.css:
```diff
@@ -443,6 +441,57 @@
+    suffix: '\1366 '; /* ፦ */
+}
+
+/* FIXME: The W3C definition calls it "haleme", check if it is a typo */
```
I'm not sure why https://w3c.github.io/predefined-counter-styles/#ethiopic-haleme calls it `Haleme`, but I could find a number of references on internet using the word `Halehame`, also referred as `Debtera` sometimes. And this is also the name used for the `list-style-type` itself. So I guess we can remove this `FIXME`. I couldn't find much with `Haleme`, so it might be good to create an spec issue, although this is in the informative part.

http://yacob.org/papers/DanielYacob-IUC19.pdf
https://www.senamirmir.org/interviews/theme/5-2001/gh/ws.html

nt1m replied:
I did file an issue here: https://github.com/w3c/predefined-counter-styles/issues/55

vitorroriz commented on Source/WebCore/css/CSSCounterStyleRegistry.cpp:
```diff
@@ -167,45 +167,6 @@ bool CSSCounterStyleRegistry::operator==(const CSSCounterStyleRegistry& other) c
     return m_authorCounterStyles == other.m_authorCounterStyles;
 }
 
-bool isCounterStyleUnsupportedByUserAgent(CSSValueID valueID)
```
You forgot to remove it from the header file.

nt1m replied:
Ah good catch, will take care of it

vitorroriz commented on Source/WebCore/style/StyleBuilderConverter.h:
```diff
@@ -296,7 +296,9 @@ inline ListStyleType BuilderConverter::convertListStyleType(const BuilderState&
     auto& primitiveValue = downcast<CSSPrimitiveValue>(value);
     if (primitiveValue.isValueID()) {
-        if (builderState.document().settings().cssCounterStyleAtRulesEnabled() && !isCounterStyleUnsupportedByUserAgent(primitiveValue.valueID()))
+        if (primitiveValue.valueID() == CSSValueNone)
```
`fromCSSValue()` should already handle `CSSValueNone`. Are you doing this explicitly as an optimization for a frequent value?

nt1m replied:
Mostly separating this out to clearly distinguish old and new codepath

nt1m committed:
[css-counter-styles] Port ethiopic counter styles to new codepath
https://bugs.webkit.org/show_bug.cgi?id=256158
rdar://108724140

Reviewed by Myles C. Maxfield.

Use converter at https://bugs.webkit.org/attachment.cgi?id=466153 to generate counter styles based on RenderListMarker data.
Note that this is supposed to be a 1:1 port from RenderListMarker, so there are places where we differ from the W3C ready-made counter styles.
rdar://108704382 has been filed to review them.

Also isCounterStyleUnsupportedByUserAgent is now removed and merged back to its call-site.

* Source/WebCore/css/CSSCounterStyleRegistry.cpp:
(WebCore::isCounterStyleUnsupportedByUserAgent): Deleted.
* Source/WebCore/css/CSSCounterStyleRegistry.h:
* Source/WebCore/css/counterStyles.css:
(@counter-style afar):
(@counter-style ethiopic-halehame-ti-er):
(@counter-style ethiopic-halehame-ti-et):
(@counter-style oromo):
(@counter-style sidama):
(@counter-style tigre):
(@counter-style urdu):
(@counter-style amharic):
(@counter-style amharic-abegede):
(@counter-style ethiopic):
(@counter-style ethiopic-abegede):
(@counter-style somali):
(@counter-style tigrinya-er-abegede):
(@counter-style tigrinya-et-abegede):
(@counter-style ethiopic-abegede-gez):
(@counter-style ethiopic-abegede-am-et):
(@counter-style ethiopic-abegede-ti-er):
(@counter-style ethiopic-abegede-ti-et):
(@counter-style ethiopic-halehame-aa-et):
(@counter-style ethiopic-halehame-aa-er):
(@counter-style ethiopic-halehame-am-et):
(@counter-style ethiopic-halehame-gez):
(@counter-style ethiopic-halehame-om-et):
(@counter-style ethiopic-halehame-sid-et):
(@counter-style ethiopic-halehame-so-et):
(@counter-style ethiopic-halehame-tig):
(@counter-style tigrinya-er):
(@counter-style tigrinya-et):
* Source/WebCore/style/StyleBuilderConverter.h:
(WebCore::Style::BuilderConverter::convertListStyleType):

Canonical link: https://commits.webkit.org/263602@main
